### PR TITLE
pylint all python script under ./jenkins

### PR DESCRIPTION
**Pull request by krzyzacy**
namely bootstrap_test.py

one step closer to #2703

### CODE REVIEW

**Comment by krzyzacy** [comment]
/unassign
/assign @fejta 

that concludes my pre-holiday fun

File: jenkins/bootstrap_test.py [hunk 1]
```diff
@@ -78,65 +81,68 @@ def __call__(self, cmd, *a, **kw):
...
             done = bootstrap.read_all(self.ended, self, lines.append)
 
         self.assertFalse(done)
         self.assertFalse(lines)
 
-    def testRead_End(self):
+    def test_read_end(self):
         """Note we reached the end of the stream."""
         lines = []
-        self.readline = lambda: ''
```
Comment (fejta) [inline jenkins/bootstrap_test.py]: Is this change necessary?

File: jenkins/bootstrap_test.py [hunk 2]
```diff
@@ -192,82 +198,85 @@ def testTerminate_Kill(self):
...
-    def testCheckFalse(self):
+    def test_check_false(self):
         """Never raise when check is not set."""
-        bootstrap._call(0, FAIL, check=False)
-        bootstrap._call(0, PASS, check=False)
+        try:
+            bootstrap._call(0, FAIL, check=False)
+            bootstrap._call(0, PASS, check=False)
+        except subprocess.CalledProcessError:
+            self.fail('Should not raise when check is not set')
```
Comment (fejta) [inline jenkins/bootstrap_test.py]: What is this about?

File: jenkins/bootstrap_test.py [hunk 3]
```diff
@@ -192,82 +198,85 @@ def testTerminate_Kill(self):
...
+    def test_pull_numbers(self):
         self.assertListEqual(bootstrap.pull_numbers(123), ['123'])
         self.assertListEqual(bootstrap.pull_numbers('master:abcd'), [])
         self.assertListEqual(
             bootstrap.pull_numbers('master:abcd,123:qwer,124:zxcv'),
             ['123', '124'])
 
-    def testPullRef(self):
+    def test_pull_ref(self):
         self.assertEqual(bootstrap.pull_ref('master:abcd,123:effe'),
```
Comment (fejta) [inline jenkins/bootstrap_test.py]: Please break the line after assertEqual( (here and in test_branch_ref)

File: jenkins/bootstrap_test.py [hunk 4]
```diff
@@ -408,58 +418,57 @@ def testDeprecatedPullRepoCommit(self):
...
         """--repo=foo=branch."""
         self.assertEquals(
             {'foo': ('this', '')},
             bootstrap.parse_repos(
                 FakeArgs(repo=['foo=this'], branch='', pull='')))
 
-    def testBranchCommit(self):
-        """--repo=foo=branch:commit works"""
+    def test_branch_commit(self):
+        """--repo=foo=branch:commit is not allowed."""
```
Comment (fejta) [inline jenkins/bootstrap_test.py]: It sure looks like --repo=foo=branch:commit is allowed to me?

File: jenkins/bootstrap_test.py [hunk 5]
```diff
@@ -1825,25 +1833,27 @@ def CheckBootstrapYaml(self, path, check, use_json=False):
...
+                self.load_bootstrap_yaml(yamlf)
+            self.load_prow_yaml(self.prow_config)
         self.assertIn(key, sorted(self.realjobs))  # sorted for clearer error message
         return self.realjobs.get(key)
 
-    def testValidTimeout(self):
+    def test_valid_timeout(self):
         """All jobs set a timeout less than 120m or set DOCKER_TIMEOUT."""
-        default_timeout = int(re.search(r'\$\{DOCKER_TIMEOUT:-(\d+)m', open('%s/dockerized-e2e-runner.sh' % os.path.dirname(__file__)).read()).group(1))
+        default_timeout = int(re.search(r'\$\{DOCKER_TIMEOUT:-(\d+)m',
```
Comment (fejta) [inline jenkins/bootstrap_test.py]: Please break the the line after a ( (ether int( or re.search( so that each args indent doesnt' depend on the length of the previous line

File: jenkins/bootstrap_test.py [hunk 6]
```diff
@@ -2013,12 +2024,13 @@ def testEnvsNoExport(self):
                     'CHARTS_TEST=',
                     'PERF_TESTS='
                 ]
-                for b in black:
-                    if b in line:
-                        self.fail('[%r]: Env %r: Convert %r to use e2e scenario flags' % (job, line, b))
+                for bad in bads:
+                    if bad in line:
+                        self.fail('[%r]: Env %r: Convert %r to '
```
Comment (fejta) [inline jenkins/bootstrap_test.py]: Please fit the literal onto one line, aka:
```
self.fail(
    'foo %s' % (
        bar, other))
```

File: jenkins/clean_project.py [hunk 7]
```diff
@@ -22,68 +24,70 @@
...
-      continue
-    name, zone, created_str = re.split(r'\s+', inst)
-    tfmt = 'YYYY-mm-ddTHH:MM:SS'
-    created = datetime.datetime.strptime(created_str[:len(tfmt)], '%Y-%m-%dT%H:%M:%S')
-    if created < age:
-      zones[zone].append(name)
-  return zones
-
+def list_instances(project, age, filt):
+    """List instances."""
```
Comment (fejta) [inline jenkins/clean_project.py]: World's most worthless pydoc :grin:  (no need to fix though)

File: verify/verify-pylint.sh [hunk 8]
```diff
@@ -22,9 +22,7 @@ export PATH="${TEST_SRCDIR}/pylint:${PATH}"
 
 # TODO(fejta): all python files
 pylint scenarios/*.py
-pylint jenkins/bootstrap.py
-pylint jenkins/check_projects.py
-pylint jenkins/janitor.py
+pylint jenkins/*.py
```
Comment (fejta) [inline verify/verify-pylint.sh]: woo!

File: jenkins/attach_agent.py [hunk 9]
```diff
@@ -14,10 +14,12 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+"""Attach Jenkins agents."""
 
 import ConfigParser
 import sys
 
+# pylint: disable=import-error
```
Comment (krzyzacy) [inline jenkins/attach_agent.py]: jenkinsapi is only available on jenkins machines

File: jenkins/bootstrap_test.py (hunk 2 above)
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: ehh, I think that's a `self-not-used`

File: jenkins/clean_project.py (hunk 7 above)
Comment (krzyzacy) [inline jenkins/clean_project.py]: I think bowei hates it more than I do and already promptly ignored all the `missing-doc-string` :-)

File: jenkins/bootstrap_test.py (hunk 1 above)
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: the err is `An attribute defined in bootstrap_test line 137 hides this method (method-hidden)`

File: jenkins/bootstrap_test.py (hunk 6 above)
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: done

File: jenkins/bootstrap_test.py (hunk 5 above)
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: done

File: jenkins/bootstrap_test.py (hunk 4 above)
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: merge is easy to screw :-( done

File: jenkins/bootstrap_test.py (hunk 3 above)
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: done

File: jenkins/attach_agent.py (hunk 9 above)
Comment (fejta) [inline jenkins/attach_agent.py]: Hmm... Can you put a TODO to fix this? We sould be able to create a py_binary bazel target that has the jenkinsapi dep. Alternatively should we add this to the image we use for running the verify check (like we do requests)?

File: jenkins/bootstrap_test.py (hunk 1 above)
Comment (fejta) [inline jenkins/bootstrap_test.py]: No, each test method gets its own instance of ReadAllTest()

File: jenkins/bootstrap_test.py (hunk 2 above)
Comment (fejta) [inline jenkins/bootstrap_test.py]: In that case:
```
@staticmethod
def test_check_false():
  pass
```

File: jenkins/bootstrap_test.py (hunk 1 above)
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: IMO it's better than disable `method-hidden` here :-)

File: jenkins/bootstrap_test.py (hunk 2 above)
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: done

File: jenkins/attach_agent.py (hunk 9 above)
Comment (krzyzacy) [inline jenkins/attach_agent.py]: added a todo

Comment (fejta) [inline jenkins/attach_agent.py]: a) Can you put the disable just on the `from jenkinsapi` line instead of the whole file?
b) Can you check with the rest of our team (Joe/Ryan/Cole) about how to fix this?

Comment (krzyzacy) [inline jenkins/attach_agent.py]: roger
